File: Hackerrank/30-Days/day-26/solution.js
function processData(input) {
  //Enter your code here
  var data = input.split("\n");
  
  var return_date = convertToDate(data[0]);
  var due_date = convertToDate(data[1]);
  
  var fine = calculateFine(return_date,due_date);
  console.log(fine);
}

function convertToDate(line){
  var data = line.split(" ");
  var date = new Date(Date.parse(`${data[1]}-${data[0]}-${data[2]}`));
  //console.log(date);
  return date;
}

function calculateFine(return_date,due_date){
  if (due_date >= return_date)
      return 0;
  
  if (return_date.getFullYear() - due_date.getFullYear() > 0)
      return 10000;
  
  var diff_time = return_date.getTime() - due_date.getTime(); 
  //console.log(diff_time);
  
  var diff_days = diff_time / (1000 * 60 * 60 * 24);
  //console.log('day:',diff_days);
  var diff_months = diff_time / (1000 * 60 * 60 * 24 * 30);
  //console.log('month:',diff_months);
  var diff_years = diff_time / (1000 * 60 * 60 * 24 * 30 * 12);
  //console.log('year:',diff_years);
  
  if (parseInt(diff_years, 10) > 0){
      return 10000;
  } else if(parseInt(diff_months, 10) > 0){
      var months = parseInt(diff_months, 10);
      return months * 500;
  } else {
      return diff_days * 15;
  }
  
  return -1;
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
  _input += input;
});

process.stdin.on("end", function () {
 processData(_input);
});

File: Hackerrank/30-Days/day-26/solution.java
import java.io.*;
import java.util.*;

public class Solution {

    public static void main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */
        Scanner scan = new Scanner(System.in); 
        String return_date = scan.nextLine();
        String due_date = scan.nextLine();
        
        int fine = calculateFine(return_date,due_date);
        System.out.println(fine);
    }
    
    public static int calculateFine(String return_date, String due_date){
        int[] return_data = convertData(return_date);
        int[] due_data = convertData(due_date);
        //System.out.println(Arrays.toString(return_data));
        int years = return_data[2] - due_data[2];
        int months = return_data[1] - due_data[1];
        int days = return_data[0] - due_data[0];
        
        if (years > 0){
            return 10000;
        } else if (months > 0 && years == 0){
            return months * 500;
        } else if (days > 0 && months == 0 && years == 0){
            return days * 15;
        }
        
        return 0;
    }
    
    public static int[] convertData(String data){
        //https://stackoverflow.com/questions/18838781/converting-string-array-to-an-integer-array
        int[] date_int = Arrays.stream(data.split(" ")).mapToInt(Integer::parseInt).toArray();
        return date_int;
    }
}